### DISCUSSION TITLE
Using WiX 3.11 to create installer that installs different modules that may be uninstalled as a whole

### DISCUSSION THREAD

**Discussion by JETMikeMc** [wixtoolset/issues#8821] [Questions] [answered]
I have been using a WiX installer project that basically bundles installers from other projects in Visual Studio 2022. I want to change this behavior and have the installer install everything from a single bundle.wxs or product.wxs.

Do I use a or ?

Are each of the "modules" to install encapsulated in a or a ?

I am not an expert WiX by any means, but have put together a bootstrap bundle installer. I want to get away from this so that all of the modules are installed and may be uninstalled as a whole and that the individual parts are not visible in Windows Add/Remove programs.

I must use WiX v3 and not any other installers like InstallShield, etc.

I guess I am not sure of the order of the different WiX sections.

I have tried the following order

<Wix>
    <Bundle>
        <BootstrapperApplicationRef>
            ....
        </BootstrapperApplicationRef>

        <Chain>
            <Fragment>
                .... module 1
            </Fragment>

            <Fragment>
                .... module 2
            </Fragment>

        </Chain>
    </Bundle>
</Wix>
I get errors saying the Bundle element contains an unexpected child element 'Fragment'

A Bundle element must have at least one child element of type Chain

**Comment by robmen** [comment]
You are missing some fundamentals of how WiX works. I'd recommend going through the [Tutorial](https://www.firegiant.com/docs/wix/tutorial/) and/or [Deployment Dojo](https://robmensching.com/deployment-dojo/episodes/s1/). When you have a broader foundation, you'll probably have better luck going forward.

**Reply by JETMikeMc** [reply]
Robmen, thank you for the information. The tutorial link that you sent me is specific to newer versions of WiX (WiX Toolset v4 and v5 Tutorial). I am on Wix 3.1x and do not plan on (at this time) updating to a newer version.  The Deployment Dojo is also specific to newer versions of WiX which does not help me really here.

**Reply by robmen** [reply, marked as answer]
Contrary to what you've read on the internet, the fundamentals of the WiX Toolset have not changed significantly from v5/v4 to v3.

Oh, there is also the very old v3 tutorial on firegiant.com still. I forget about that because WiX v3 goes out of community service in 3 months and is pretty far in my rearview mirror for non-customers.

**Reply by JETMikeMc** [reply]
I still am going to stick with WiX 3.11 for now as I don't want to widen the scope and work needed to change existing installers to v5/v4.  This I may do for my next version of the project. As for getting an installer to install 3 or 4 separate products in one master installer, I am still looking for a solution. I thought my original post explained the problem.

**Reply by robmen** [reply]
> I thought my original post explained the problem.

It did. Your understanding of how Fragments work is pretty far from how they actually work. That's why I suggested going back to fundamentals.

**Comment by JETMikeMc** [comment]
Thank Rob. I will look into this.
Mike



[cid:4b9cb6fc-afc3-4458-a718-1a34f4e6c0ff]
________________________________
